File: app/src/main/java/com/routesme/taxi_screen/java/Server/Interface/RoutesApi.java
package com.routesme.taxi_screen.java.Server.Interface;



import com.routesme.taxi_screen.kotlin.MVVM.Model.Institutions;
import com.routesme.taxi_screen.kotlin.MVVM.Model.RegistrationCredentials;
import com.routesme.taxi_screen.kotlin.MVVM.Model.RegistrationSuccessResponse;
import com.routesme.taxi_screen.kotlin.MVVM.Model.Vehicles;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

public interface RoutesApi {

    //Taxi Office List
    @GET("institutions")
    Call<Institutions> getInstitutions(@Query("offset") int offset, @Query("limit") int limit);
    //Taxi Offic plates list
    @GET("vehicles")
    Call<Vehicles> getVehicles(@Query("offset") int offset, @Query("limit") int limit, @Query("institutionId") int institutionId);
    //Tablet Register
    @POST("devices")
    Call<RegistrationSuccessResponse> postDevice(@Body RegistrationCredentials registrationCredentials);
}

File: app/src/main/java/com/routesme/taxi_screen/kotlin/Server/RoutesApi.kt
package com.routesme.taxi_screen.kotlin.Server

import com.google.gson.JsonElement
import com.routesme.taxi_screen.kotlin.MVVM.Model.SignInCredentials
import com.routesme.taxi_screen.kotlin.Model.*
import retrofit2.Call
import retrofit2.http.*

interface RoutesApi {


    @POST("v1/signin")
    fun signIn(@Body signInCredentials: SignInCredentials?): Call<JsonElement>

    @POST("devices")
    fun register(@Body registrationCredentials: RegistrationCredentials?): Call<JsonElement>

    //Advertisements...
    @GET("Channels")
    fun getVideos(@Query("channelidvideolist") ch_ID_Videos: Int): Call<List<VideoModel>>

    @GET("Channels")
    fun getBanners(@Query("channelidadvlist") ch_ID_Banners: Int): Call<List<BannerModel>>

    @GET("Contents")
    fun getContent(): Call<ContentResponse>
}